server/desolver.ts

```diff
@@ -32,57 +33,87 @@ export class Desolver {
   }
 
   private hasNext: number = 0;
-  private pipeline: Resolver[];
-  private resolvedObject: ResolvedObject = { resolved: false, value: null }
+  private pipeline!: Resolver[];
+  private escapeDesolver: EscapeDesolver = {wasCalled: false, resolveVal: null}
 
   constructor(
     public parent: Record<string, unknown>,
     public args: Record<string, unknown>,
     public context: Record<string, unknown>,
-    public info: Record<string, unknown>
+    public info: Record<string, unknown>,
   ) {
     this.next = this.next.bind(this);
+    this.escapeHatch = this.escapeHatch.bind(this)
+  }
+
+  // Consider refactoring the below using the 'cause' proptery in custom error types
+  // Consider own Error class to differentiate errors? Is this needed?  
+  public errorLogger(error: any): any {
+    let errorObj = {
+      'Error': error.toString(),
+      'Error Name': error.name,
+      'Error Message': error.message,
+    }
+    throw new Error(`failed to resolve ${this.pipeline[this.hasNext]}: ${errorObj}`)
+    // ^ how can I refacor the above to include multiple error parameters, Error(message, options)
   }
 
   public composePipeline(...resolvers: Resolver[]): unknown {
     this.pipeline = resolvers;
     return this.execute();
   }
 
-  private execute(): unknown {
-    while (this.hasNext <= this.pipeline.length - 1) {
-      if (this.resolvedObject.resolved) return this.resolvedObject.value;
-
-      if (this.hasNext === this.pipeline.length - 1) {
-        return this.pipeline[this.hasNext](
+  private async execute(): Promise<unknown> {
+    while (this.hasNext < this.pipeline.length - 1) {
```

Comment:
This line here needs a <= instead of <.  Because of this, the Desolver pipeline escapeHatch doesn't work if you pass a single function in but will work with multiple functions.

Reply:
Thank you, line 67 has been updated to include <=